packages/jest-worker/__typetests__/jest-worker.test.ts

```diff
@@ -5,96 +5,98 @@
+  expect(unknownWorkerFarm).type.not.toHaveProperty('isResult');
 
-expectType<Promise<{forceExited: boolean}>>(unknownWorkerFarm.end());
-expectType<NodeJS.ReadableStream>(unknownWorkerFarm.getStderr());
-expectType<NodeJS.ReadableStream>(unknownWorkerFarm.getStdout());
+  expect(unknownWorkerFarm).type.not.toHaveProperty('setup');
+  expect(unknownWorkerFarm).type.not.toHaveProperty('teardown');
 
-// detected JestWorkerFarm
+  expect(unknownWorkerFarm.start()).type.toEqual<Promise<void>>();
+  expect(unknownWorkerFarm.end()).type.toEqual<
+    Promise<{forceExited: boolean}>
+  >();
 
-declare const detectedWorkerFarm: JestWorkerFarm<typeof testWorker>;
+  expect(unknownWorkerFarm.getStderr()).type.toEqual<NodeJS.ReadableStream>();
+  expect(unknownWorkerFarm.getStdout()).type.toEqual<NodeJS.ReadableStream>();
+});
 
-expectType<Promise<void>>(detectedWorkerFarm.runTest('abc', true));
-expectType<Promise<void>>(detectedWorkerFarm.runTestAsync(123, 456));
+declare const inferredWorkerFarm: JestWorkerFarm<typeof testWorker>;
 
-expectType<Promise<void>>(detectedWorkerFarm.doSomething());
-expectType<Promise<void>>(detectedWorkerFarm.doSomething());
-expectType<Promise<void>>(detectedWorkerFarm.doSomethingAsync());
-expectType<Promise<void>>(detectedWorkerFarm.doSomethingAsync());
+test('inferred JestWorkerFarm', () => {
+  expect(inferredWorkerFarm.runTest('abc', true)).type.toEqual<Promise<void>>();
+  expect(inferredWorkerFarm.runTestAsync(123, 456)).type.toEqual<
+    Promise<void>
+  >();
 
-expectError(detectedWorkerFarm.runTest());
-expectError(detectedWorkerFarm.runTest('abc'));
-expectError(detectedWorkerFarm.runTestAsync());
-expectError(detectedWorkerFarm.runTestAsync(123));
-expectError(detectedWorkerFarm.doSomething(123));
-expectError(detectedWorkerFarm.doSomethingAsync('abc'));
+  expect(inferredWorkerFarm.doSomething()).type.toEqual<Promise<void>>();
+  expect(inferredWorkerFarm.doSomething()).type.toEqual<Promise<void>>();
+  expect(inferredWorkerFarm.doSomethingAsync()).type.toEqual<Promise<void>>();
+  expect(inferredWorkerFarm.doSomethingAsync()).type.toEqual<Promise<void>>();
 
-expectError(detectedWorkerFarm.getResult());
-expectError(detectedWorkerFarm.isResult);
+  expect(inferredWorkerFarm.runTest()).type.toRaiseError();
+  expect(inferredWorkerFarm.runTest('abc')).type.toRaiseError();
+  expect(inferredWorkerFarm.runTestAsync()).type.toRaiseError();
+  expect(inferredWorkerFarm.runTestAsync(123)).type.toRaiseError();
+  expect(inferredWorkerFarm.doSomething(123)).type.toRaiseError();
+  expect(inferredWorkerFarm.doSomethingAsync('abc')).type.toRaiseError();
 
-expectError(detectedWorkerFarm.setup());
-expectError(detectedWorkerFarm.teardown());
+  expect(inferredWorkerFarm).type.not.toHaveProperty('getResult');
+  expect(inferredWorkerFarm).type.not.toHaveProperty('isResult');
 
-expectNotAssignable<Promise<void>>(detectedWorkerFarm.end());
-expectType<Promise<{forceExited: boolean}>>(detectedWorkerFarm.end());
```

Comment:
It does not make sense to check assignability, when equality is checked in the next line. If the resulting type if `Promise<{forceExited: boolean}>` it is obviously not assignable to `Promise<void>`. Looks redundant. I removed assignability checks here and in the next test.